build/preprocessing.py

```diff
@@ -0,0 +1,72 @@
+#! python
+
+import yaml
+import sys
+from os import listdir
+from os.path import isfile, join
+
+if __name__ == "__main__":
+
+  print("Running preprocessing for common templates")
+
+  rules = None
+  patchPath = sys.argv[1] if len(sys.argv) > 1 else "patch.yaml"
+  with open(patchPath, 'r') as stream:
+    try:
+      patch = yaml.load(stream, Loader=yaml.BaseLoader)
+
+      if patch == None: 
+        print("Nothing in patch file")
+        sys.exit(0)
+      
+      rules = patch.get("rules")
+      for rule in rules:
+        rule["matchLabel"] = rule.get("matchLabel").split(": ")
+        rule["addAnnotation"] = rule.get("addAnnotation").split(": ")
+    except yaml.YAMLError as exc:
+      print(exc)
+      sys.exit(1)
+
+  commonTemplatesPath = sys.argv[2] if len(sys.argv) > 2 else "/opt/ansible/roles/KubevirtCommonTemplatesBundle/files"
+  files = [f for f in listdir(commonTemplatesPath) if isfile(join(commonTemplatesPath, f))]
+
+  for fileName in files:
+
+    commonTemplates = None
+    with open(commonTemplatesPath+fileName, 'r') as stream:
+      try:
+        commonTemplates = list(yaml.safe_load_all(stream))
+        if len(commonTemplates) == 0: 
+          print("Nothing in common templates file")
+          continue
+
+      except yaml.YAMLError as exc:
+        print(exc)
+        continue
+
+    print("------------------------------------------------------")
+    print("Running script for file: " + fileName)
+    print("------------------------------------------------------")
```

Comment:
the processing steps are well defined enough to fit each in a separate function. This would be a nice improvement, but the biggest gain is to move all the application code in a function, to make it easily testable.